src/NewProject/source_grid.cpp

```diff
@@ -540,12 +550,54 @@ QString sourceGrid::Filter(int projectType, GridType gType) const {
   return DESIGN_SOURCES_FILTER;
 }
 
-void sourceGrid::VerifyFilesWithSameName(const QStringList &files) {
+bool sourceGrid::VerifyFilesWithSameName(const QStringList &files,
+                                         bool override) {
   if (!files.isEmpty()) {
-    QMessageBox::warning(
-        this, "File(s) already added",
-        QString{"The folowing files already added to project.\n\n%1"}.arg(
-            files.join("\n")));
+    if (override) {
+      const int returnCode = QMessageBox::warning(
+          this, "File already added",
+          QString{"File %1 has already been added to the project"
+                  ". Do you want to override it?"}
+              .arg(files.first()),
+          QMessageBox::Yes | QMessageBox::No | QMessageBox::Cancel);
+      return returnCode == QMessageBox::Yes;
+    } else {
+      QMessageBox::warning(this, "File(s) already added",
```

Comment:
```suggestion
      QMessageBox::warning(this, "File overwrite",
```

Reply:
> 

This is old message when you add few files or directory. Nothing was changed for that cases.